onebusaway-android/src/main/java/org/onebusaway/android/ui/TripDetailsListFragment.java

```diff
@@ -0,0 +1,352 @@
+
+    private static final int TRIP_DETAILS_LOADER = 0;
+
+    private String mTripId;
+
+    private String mStopId;
+
+    private ObaTripDetailsResponse mTripInfo;
+
+    private ListAdapter mAdapter;
+
+    private final TripDetailsLoaderCallback mTripDetailsCallback = new TripDetailsLoaderCallback();
+
+    @Override
+    public void onActivityCreated(Bundle savedInstanceState) {
+        super.onActivityCreated(savedInstanceState);
+
+        // Start out with a progress indicator.
+        setListShown(false);
+
+        getListView().setOnItemClickListener(mClickListener);
+
+        // Get the tripID from the "uri" segment
+        Uri uri = getArguments().getParcelable(FragmentUtils.URI);
+        if (uri == null) {
+            Log.e(TAG, "No URI in arguments");
+            return;
+        }
+
+        extractTripIdAndStopId(uri);
+
+        getLoaderManager().initLoader(TRIP_DETAILS_LOADER, null, mTripDetailsCallback);
+    }
+
+    private void extractTripIdAndStopId(Uri uri) {
+        // get the segments after 'trips'
+        List<String> segments = uri.getPathSegments();
+        int offsetFromTripsSegment = -1;
+        for (String segment : segments) {
+            if (offsetFromTripsSegment == -1) {
+                if (segment.equals(ObaContract.Trips.PATH)) {
+                    offsetFromTripsSegment = 0;
+                }
+            } else {
+                offsetFromTripsSegment++;
+                if (offsetFromTripsSegment == 1) {
+                    mTripId = segment;
+                } else if (offsetFromTripsSegment == 2) {
+                    mStopId = segment;
+                }
+            }
+        }
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater,
+                             ViewGroup root, Bundle savedInstanceState) {
+        if (root == null) {
+            // Currently in a layout without a container, so no
+            // reason to create our view.
+            return null;
+        }
+        return inflater.inflate(R.layout.trip_details, null);
+    }
+
+    private void setTripDetails(ObaTripDetailsResponse data) {
+        mTripInfo = data;
+
+        final int code = mTripInfo.getCode();
+        if (code == ObaApi.OBA_OK) {
+            setEmptyText("");
+        } else {
+            setEmptyText(UIHelp.getRouteErrorString(getActivity(), code));
+            return;
+        }
+
+        setUpHeader();
+
+        mAdapter = new TripDetailsAdapter();
+
+        setListAdapater(mAdapter);
+
+        if (mStopId != null) {
+            final int stopIndex = findIndexForStop(mTripInfo.getSchedule().getStopTimes(), mStopId);
+            if (stopIndex != -1) {
+                final ListView listView = getListView();
+                listView.post(new Runnable() {
+                    @Override
+                    public void run() {
+                        listView.smoothScrollToPosition(stopIndex);
```

Comment:
The smooth scroll looks really strange to me - let's change to just setting the position to the `ListView` snaps to that position without scrolling.  IIRC, `listview.setSelection()` is the method that does this.